validator/client/beacon-api/activation.go

```diff
@@ -0,0 +1,158 @@
+//go:build use_beacon_api
+// +build use_beacon_api
+
+package beacon_api
+
+import (
+	"context"
+	"encoding/hex"
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"net/url"
+	"strconv"
+
+	"github.com/pkg/errors"
+	"github.com/prysmaticlabs/prysm/v3/api/gateway/apimiddleware"
+	types "github.com/prysmaticlabs/prysm/v3/consensus-types/primitives"
+	ethpb "github.com/prysmaticlabs/prysm/v3/proto/prysm/v1alpha1"
+	"google.golang.org/grpc"
```

Comment:
Is Bazel able to build with this new import? It's not in the `BUILD.bazel` file.

Reply:
Actually I built it directly in VSCode.
There is a strong possibility it bypasses Bazel.
Let me check it.

Reply:
Is there an easy way to automatically add dependancies?

[Documentation](https://docs.prylabs.network/docs/reading/bazel#do-i-have-to-edit-build-files-myself) states: 
```
bazel run //:gazelle -- fix
```
On develop branch, it shows:

```
INFO: Analyzed target //:gazelle (0 packages loaded, 0 targets configured).
INFO: Found 1 target...
INFO: From BazelWorkspaceStatusAction stable-status.txt:
date: illegal option -- -
usage: date [-jnRu] [-r seconds|file] [-v[+|-]val[ymwdHMS]]
            [-I[date | hours | minutes | seconds]]
            [-f fmt date | [[[mm]dd]HH]MM[[cc]yy][.ss]] [+format]
date: illegal option -- -
usage: date [-jnRu] [-r seconds|file] [-v[+|-]val[ymwdHMS]]
            [-I[date | hours | minutes | seconds]]
            [-f fmt date | [[[mm]dd]HH]MM[[cc]yy][.ss]] [+format]
Target //:gazelle up-to-date:
  bazel-bin/gazelle-runner.bash
  bazel-bin/gazelle
INFO: Elapsed time: 3.654s, Critical Path: 0.04s
INFO: 1 process: 1 internal.
INFO: Build completed successfully, 1 total action
INFO: Build completed successfully, 1 total action
gazelle: /Users/manu/Kiln/prysm/beacon-chain/state/state-native/BUILD.bazel:5.12-42.7: could not merge expression
gazelle: /Users/manu/Kiln/prysm/cmd/validator/flags/BUILD.bazel:5.12-11.7: could not merge expression
gazelle: /Users/manu/Kiln/prysm/config/fieldparams/BUILD.bazel:5.12-8.7: could not merge expression
gazelle: /Users/manu/Kiln/prysm/validator/client/validator-client-factory/BUILD.bazel:18.12-21.7: could not merge expression
gazelle: /Users/manu/Kiln/prysm/validator/client/validator-client-factory/BUILD.bazel:24.12-30.7: could not merge expression
```

Reply:
Added dependencies by hand

Reply:
I think gazelle does not work here because of this:
```
# Gazelle will flag this for removal since it's only being used with the use_beacon_api flag, but it should be kept
# keep
```